Columns must be same length as key

```text
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/sr-detection/sr-detection/sr-data/src/sr_data/steps/gh_mentions.py", line 36, in main
    frame[["readme_pmentions", "readme_imentions"]] = (
    ~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/pypoetry/virtualenvs/sr-data-iIS7NBSl-py3.12/lib/python3.12/site-packages/pandas/core/frame.py", line 4299, in __setitem__
    self._setitem_array(key, value)
  File "/home/runner/.cache/pypoetry/virtualenvs/sr-data-iIS7NBSl-py3.12/lib/python3.12/site-packages/pandas/core/frame.py", line 4358, in _setitem_array
    self._iset_not_inplace(key, value)
  File "/home/runner/.cache/pypoetry/virtualenvs/sr-data-iIS7NBSl-py3.12/lib/python3.12/site-packages/pandas/core/frame.py", line 4377, in _iset_not_inplace
    raise ValueError("Columns must be same length as key")
ValueError: Columns must be same length as key
error: Recipe `ghmentions` failed on line 122 with exit code 1
cp: cannot stat 'after-ghmentions.csv': No such file or directory
Error: Process completed with exit code 1.
```

--- a/sr-data/src/sr_data/steps/gh_mentions.py
+++ b/sr-data/src/sr_data/steps/gh_mentions.py
@@ -33,8 +33,12 @@ def main(repos, out):
     logger.info(
         f"Counting GitHub pull request, and issue mentions in {len(frame)} repositories"
     )
-    frame[["readme_pmentions", "readme_imentions"]] = (
-        frame["readme"].apply(mentions)).apply(pd.Series)
+    if frame.empty:
+        frame["readme_pmentions"] = 0
+        frame["readme_imentions"] = 0
+    else:
+        frame[["readme_pmentions", "readme_imentions"]] = (
+            frame["readme"].apply(mentions)).apply(pd.Series)
     frame.to_csv(out, index=False)
     logger.info(f"Saved {len(frame)} repositories to {out}")
 

--- a/sr-data/src/tests/resources/to-ghmentions-empty.csv
+++ b/sr-data/src/tests/resources/to-ghmentions-empty.csv
@@ -1,0 +1,1 @@
+repo,readme

--- a/sr-data/src/tests/test_ghmentions.py
+++ b/sr-data/src/tests/test_ghmentions.py
@@ -79,3 +79,17 @@ class TestGhMentions(unittest.TestCase):
             frame = pd.read_csv(path)
             self.assertEqual(frame.iloc[0]["readme_imentions"], 1)
             self.assertEqual(frame.iloc[0]["readme_pmentions"], 1)
+
+    @pytest.mark.fast
+    def test_returns_zero_records_on_empty_input(self):
+        with TemporaryDirectory() as temp:
+            path = os.path.join(temp, "mentions.csv")
+            main(
+                os.path.join(
+                    os.path.dirname(os.path.realpath(__file__)),
+                    "resources/to-ghmentions-empty.csv"
+                ),
+                path
+            )
+            frame = pd.read_csv(path)
+            self.assertTrue(frame.empty)

